### PR TITLE
Replacing deprecated Gradle config 'compile' with 'api' and 'implementation'

### PR DESCRIPTION
**Pull request by bhariharan**

### CODE REVIEW

File: hybrid/HybridSampleApps/SmartSyncExplorerHybrid/build.gradle [hunk 1]
```diff
@@ -1,7 +1,7 @@
 apply plugin: 'com.android.application'
 
 dependencies {
-  compile project(':libs:SalesforceHybrid')
+  implementation project(':libs:SalesforceHybrid')
```
Comment (bhariharan) [inline hybrid/HybridSampleApps/SmartSyncExplorerHybrid/build.gradle]: `implementation` for our libraries that have sources.

File: libs/SalesforceAnalytics/build.gradle [hunk 2]
```diff
@@ -14,7 +14,7 @@ apply plugin: 'com.github.dcendents.android-maven'
 apply plugin: 'com.jfrog.bintray'
 
 dependencies {
-    compile 'com.squareup:tape:1.2.3'
+    api 'com.squareup:tape:1.2.3'
```
Comment (bhariharan) [inline libs/SalesforceAnalytics/build.gradle]: `api` for third-party libraries that may or may not have sources and are pulled from `Maven`.

**Comment by brandonpage** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please provide a summary in the Pull Request description</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_ci/circleci">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by bhariharan** [comment]
I got that wrong - everything should use `api`. A good read of the difference between `implementation` and `api` is [here](https://blog.mindorks.com/implementation-vs-api-in-gradle-3-0-494c817a6fa).

**Pull request merged**